_posts/2015-02-06-Utilising-API-Blueprint-in-API-Clients.md

```diff
@@ -0,0 +1,119 @@
+layout: post
+date: 2015-02-06 18:00:00 +0000
+author: kyle
+published: true
+comments: true
+---
+
+I’m excited to release a new way to interface to API Blueprints. A client-side framework in Swift that allows you to introspect and perform actions at run-time.
+
+Imagine an API client when you specify which resource and action you want to perform from an API Blueprint, it can use the already defined information from a blueprint to build the request without having to duplicate the information such as the HTTP method and URI from your blueprint. This allows you to start building smarter clients which can evolve at run-time based on the blueprint.
+
+The SDK is built on-top of a framework we’ve built called the [Representor](https://github.com/the-hypermedia-project/charter), which is designed to be a canonical representation of a hypermedia message in multiple languages. One of the key constraints to Representational State Transfer (REST) is Hypermedia as the Engine of State (HATEOS).  The message from the server itself should contain information about how to transition to other states along with links to other resources.
+
+The Representor is a structure providing this functionality along with the ability to deserialise Hypermedia formats such as [HAL](http://stateless.co/hal_specification.html) and [Siren](https://github.com/kevinswiber/siren). The Representer contains a couple of properties, one of them is it’s transitions to other states, along with how the current resource links to other resources.
+
+Since a blueprint is effectively a document which specifies all of the transitions (actions) in your API, we’ve built on-top of the Representor and added the ability to deserialise an API Blueprint into a collection of transitions.
+
+This allows us to decouple an API client from the server-side implementation, we no longer have to know about the URIs to each resource and the HTTP methods in our code. We can negotiate these at run-time, similar to how with HATEOS we follow links, we can follow links from an API Blueprint.
+
+## Example
+
+Let’s walk though how to load an API Blueprint and perform transitions in Swift. First of all, we’re going to load our blueprint file that’s embedded in our application. You’re not limited to embedding it in your application. You could even retrieve it dynamically at run-time.
+
+```swift
+let blueprint = Blueprint(named: "blueprint.json", bundle: nil)
+```
+
+Once you’ve loaded your blueprint, you can retrieve the different actions as [transitions](https://github.com/the-hypermedia-project/charter/blob/master/reference/hypermedia-elements.md#transitions) from your blueprint.
+
+This example is going to make use of one of our example blueprints, “Polls API”. I’ve pulled out an excerpt of the example blueprint and it looks something like the following. You can find the full version over in our [examples](https://github.com/apiaryio/api-blueprint/blob/kylef/polls-example/examples/Polls%20API.md).
```

Comment:
I think we should point this instead of GH to Apiary. Create a new API under Apiary team and copy paste there that example (and yes I know :((), then don't forget to add the link to GH example too (what if somebody wants to see the source). 

And yes, this is making me really sad – years I am trying to get the access to API blueprints but we still have to go through this.